From 2795d7515acfbd23723429ee5b611de88404d324 Mon Sep 17 00:00:00 2001
From: Dmitry Rozhkov <dmitry.rozhkov@linux.intel.com>
Date: Wed, 25 Jan 2017 11:06:58 +0200
Subject: [PATCH 2/2] moveit_ros/planning: Use TinyXML2 instead of TinyXML

The library TinyXML is considered to be unmaintained and
since all future development is focused on TinyXML2 this
patch updates moveit_ros_planning to use TinyXML2.

Upstream-Status: Submitted [https://github.com/ros-planning/moveit/pull/527]

Signed-off-by: Dmitry Rozhkov <dmitry.rozhkov@linux.intel.com>
---
 rdf_loader/include/moveit/rdf_loader/rdf_loader.h | 4 ++--
 rdf_loader/src/rdf_loader.cpp                     | 4 +++-
 .../include/moveit/robot_model_loader/robot_model_loader.h            | 4 ++--
 3 files changed, 7 insertions(+), 5 deletions(-)

diff --git a/rdf_loader/include/moveit/rdf_loader/rdf_loader.h b/rdf_loader/include/moveit/rdf_loader/rdf_loader.h
index a313e1b..0999963 100644
--- a/rdf_loader/include/moveit/rdf_loader/rdf_loader.h
+++ b/rdf_loader/include/moveit/rdf_loader/rdf_loader.h
@@ -41,7 +41,7 @@
 #include <urdf/model.h>
 #include <srdfdom/model.h>
 #include <boost/shared_ptr.hpp>
-#include <tinyxml.h>
+#include <tinyxml2.h>
 
 namespace rdf_loader
 {
@@ -62,7 +62,7 @@ public:
   RDFLoader(const std::string& urdf_string, const std::string& srdf_string);
 
   /** \brief Initialize the robot model from a parsed XML representation of the URDF and SRDF */
-  RDFLoader(TiXmlDocument* urdf_doc, TiXmlDocument* srdf_doc);
+  RDFLoader(tinyxml2::XMLDocument* urdf_doc, tinyxml2::XMLDocument* srdf_doc);
 
   /** @brief Get the resolved parameter name for the robot description */
   const std::string& getRobotDescription() const
diff --git a/rdf_loader/src/rdf_loader.cpp b/rdf_loader/src/rdf_loader.cpp
index a814124..0185714 100644
--- a/rdf_loader/src/rdf_loader.cpp
+++ b/rdf_loader/src/rdf_loader.cpp
@@ -38,6 +38,8 @@
 #include <moveit/profiler/profiler.h>
 #include <ros/ros.h>
 
+using namespace tinyxml2;
+
 rdf_loader::RDFLoader::RDFLoader(const std::string& robot_description)
 {
   moveit::tools::Profiler::ScopedStart prof_start;
@@ -104,7 +106,7 @@ rdf_loader::RDFLoader::RDFLoader(const std::string& urdf_string, const std::stri
   }
 }
 
-rdf_loader::RDFLoader::RDFLoader(TiXmlDocument* urdf_doc, TiXmlDocument* srdf_doc)
+rdf_loader::RDFLoader::RDFLoader(XMLDocument* urdf_doc, XMLDocument* srdf_doc)
 {
   moveit::tools::Profiler::ScopedStart prof_start;
   moveit::tools::Profiler::ScopedBlock prof_block("RDFLoader(XML)");
diff --git a/robot_model_loader/include/moveit/robot_model_loader/robot_model_loader.h b/robot_model_loader/include/moveit/robot_model_loader/robot_model_loader.h
index 988ea73..7c88c0a 100644
--- a/robot_model_loader/include/moveit/robot_model_loader/robot_model_loader.h
+++ b/robot_model_loader/include/moveit/robot_model_loader/robot_model_loader.h
@@ -67,7 +67,7 @@ public:
     {
     }
 
-    Options(TiXmlDocument* urdf_doc, TiXmlDocument* srdf_doc)
+    Options(tinyxml2::XMLDocument* urdf_doc, tinyxml2::XMLDocument* srdf_doc)
       : urdf_doc_(urdf_doc), srdf_doc_(srdf_doc), load_kinematics_solvers_(true)
     {
     }
@@ -82,7 +82,7 @@ public:
     std::string urdf_string_, srdf_string_;
 
     /** @brief The parsed XML content of the URDF and SRDF documents. */
-    TiXmlDocument *urdf_doc_, *srdf_doc_;
+    tinyxml2::XMLDocument *urdf_doc_, *srdf_doc_;
 
     /** @brief Flag indicating whether the kinematics solvers should be loaded as well, using specified ROS parameters
      */
-- 
2.7.4

